### PR TITLE
(feat) Upgrade to cython 3.0.2 (conda)

### PR DESCRIPTION
**Pull request by MementoRC**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Trying to just limit to cython 3.0.2 update.

**Tests performed by the developer**:
./clean;./compile;make test

**Tips for QA testing**:

### CODE REVIEW

File: hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_constants.py [hunk 1]
```diff
@@ -6,8 +6,6 @@
 EXCHANGE_NAME = "phemex_perpetual"
 MAX_ORDER_ID_LEN = 40
 
-HB_PARTNER_ID = "HBOT"
```
Comment (cardosofede) [inline hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_constants.py:9]: this shouldn't be removed

File: hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_derivative.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ def client_order_id_max_length(self) -> int:
 
     @property
     def client_order_id_prefix(self) -> str:
-        return CONSTANTS.HB_PARTNER_ID
```
Comment (cardosofede) [inline hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_derivative.py:90]: this neither

File: hummingbot/connector/exchange/hitbtc/hitbtc_api_order_book_data_source.py [hunk 3]
```diff
@@ -43,6 +43,7 @@ async def init_trading_pair_symbols(cls, shared_session: Optional[aiohttp.Client
             "GET",
             Constants.ENDPOINT["SYMBOL"],
             shared_client=shared_session)
+        print(symbols)
```
Comment (cardosofede) [inline hummingbot/connector/exchange/hitbtc/hitbtc_api_order_book_data_source.py:46]: this is not necessary

File: hummingbot/connector/exchange/okx/okx_auth.py [hunk 4]
```diff
@@ -61,7 +61,9 @@ def _generate_signature(self, timestamp: str, method: str, path_url: str, body:
         return signature
 
     def authentication_headers(self, request: RESTRequest) -> Dict[str, Any]:
-        timestamp = datetime.utcfromtimestamp(self.time_provider.time()).isoformat(timespec="milliseconds") + "Z"
+        timestamp = datetime.utcfromtimestamp(self.time_provider.time()).isoformat(timespec="milliseconds")
```
Comment (cardosofede) [inline hummingbot/connector/exchange/okx/okx_auth.py:64]: why we have this change? is tested?

File: hummingbot/connector/test_support/mock_paper_exchange.pyx [hunk 5]
```diff
@@ -65,7 +65,8 @@ cdef class MockPaperExchange(PaperTradeExchange):
         return True
 
     def split_trading_pair(self, trading_pair: str) -> Tuple[str, str]:
-        return trading_pair.split("-")
+        base, quote = trading_pair.split("-")
```
Comment (cardosofede) [inline hummingbot/connector/test_support/mock_paper_exchange.pyx:68]: why this change is necessary?

File: hummingbot/model/funding_payment.py [hunk 6]
```diff
@@ -73,7 +63,7 @@ def to_pandas(cls, payments: List):
             index += 1
             data.append([
                 index,
-                datetime.fromtimestamp(int(payment.timestamp / 1e3)).strftime("%Y-%m-%d %H:%M:%S"),
+                datetime.fromtimestamp(int(float(payment.timestamp) / 1e3)).strftime("%Y-%m-%d %H:%M:%S"),
```
Comment (cardosofede) [inline hummingbot/model/funding_payment.py:66]: is necessary to cast to float to make the division? was a Decimal before?

**Review by cardosofede — CHANGES_REQUESTED** [review]

File: hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_constants.py (hunk 1 above)
Comment (MementoRC) [reply hummingbot/connector/derivative/phemex_perpetual/phemex_perpetual_constants.py:9]: o.0 ... I did not intend to remove this.

File: hummingbot/connector/exchange/okx/okx_auth.py (hunk 4 above)
Comment (MementoRC) [reply hummingbot/connector/exchange/okx/okx_auth.py:64]: Ok, this did fail the tests. The `Z` conflicts with the timezone aware `isoformat()`, i.e. `YYYY-MM-HH:MM:SS+00:00`. This cannot have a `Z` at the end. It may not come from `cython` per-se, but associated updates in the .yml

Ok, I know I may not be the that good dev, but comeon ... of course it is tested. The way I test: I uninstall `hummingbot` env, install it from scratch, then clean/compile/make test ... plus does it not pass the workflow!

File: hummingbot/connector/test_support/mock_paper_exchange.pyx (hunk 5 above)
Comment (MementoRC) [reply hummingbot/connector/test_support/mock_paper_exchange.pyx:68]: This is because of the stricter conformance to typing, `trading_pair.split("-")` returns a List, `split_trading_pair()` is supposed to return a `Tuple`

File: hummingbot/model/funding_payment.py (hunk 6 above)
Comment (MementoRC) [reply hummingbot/model/funding_payment.py:66]: I think it failed the tests. It could that it was expected to be `payment.timestamp / Decimal("1e3")`

**Comment by MementoRC** [comment]
Oh well
